### PR TITLE
5005 add edit step group

### PR DESCRIPTION
**Pull request by X0nic**
Adds the ability to edit groups, as well as create child groups

### CODE REVIEW

File: app/controllers/admin/project_groups_controller.rb [hunk 1]
```diff
@@ -14,7 +19,13 @@ def create
     authorize @entry
 
     @loan = @entry.project
-    @entry.parent = @loan.root_timeline_entry
+
+    parent_id = project_group_params[:parent_id]
+    if parent_id && parent_id.present?
```
Comment (hooverlunch) [inline app/controllers/admin/project_groups_controller.rb]: Don't think you need both of these expressions. .present? checks for nil? or empty?

File: app/controllers/admin/project_groups_controller.rb [hunk 2]
```diff
@@ -14,7 +19,13 @@ def create
...
+      @entry.parent = ProjectGroup.find(parent_id)
+    else
+      @entry.parent = @loan.root_timeline_entry
+    end
```
Comment (hooverlunch) [inline app/controllers/admin/project_groups_controller.rb]: This should be just `@entry.parent = parent_id.present? ? ProjectGroup.find(parent_id) : @loan.root_timeline_entry` I think.

File: app/controllers/admin/project_groups_controller.rb [hunk 3]
```diff
@@ -23,14 +34,32 @@ def create
...
+    if @entry.update(project_group_params)
+      render nothing: true
+    else
+      render_modal_partial(status: 422)
+    end
+  end
+
+  def edit
+    render_modal_partial
+  end
```
Comment (hooverlunch) [inline app/controllers/admin/project_groups_controller.rb]: We usually try to follow the Rails scaffold method order. So ISNECUD.

File: config/locales/es/es.yml [hunk 4]
```diff
@@ -389,6 +389,7 @@ es:
 
   project_group:
     add: "Translate Me"
+    edit: "Translate Me"
```
Comment (hooverlunch) [inline config/locales/es/es.yml]: Can you please add a happy path feature spec, one for adding child and one for editing? I think we need to start doing these as part of frontend issues.

I suggest adding `  gem "capybara-screenshot", "~> 1.0"` to the gemfile. It helps a lot for debugging.

We will also need a web driver. Do you use poltergeist? If so please add it or suggest an alternative. Also can you please add PhantomJS or equivalent to the Readme.md as a dependency for dev envs.

Thank you!

File: app/assets/javascripts/backbone/views/timeline_table_view.coffee [hunk 5]
```diff
@@ -20,9 +22,18 @@ class MS.Views.TimelineTableView extends Backbone.View
 
   newGroup: (e) ->
     e.preventDefault()
-    @modal.show()
+    @modal.new()
+
+  newChildGroup: (e) ->
+    e.preventDefault()
```
Comment (X0nic) [inline app/assets/javascripts/backbone/views/timeline_table_view.coffee]: We see `e.preventDefault()` in each of these methods, but I don't know of a way to clean that up without making some questionable choices / unexpected behaviour.

**Pull request merged**